Python/chapter01/1.8 - Zero Matrix/miguel_1.8_sol.py

```diff
@@ -0,0 +1,135 @@
+"""
+Python version 3.7.0
+1.8 - Zero Matrix
+Write an algorithm such that if an element in an MxN matrix is 0, its entire row and column are set to 0
+"""
+import unittest
+from typing import List
+
+
+def zero_matrix(matrix: List[List[int]]) -> List[List[int]]:
+    """
+    zero_matrix will take in an MxN matrix and when a 0 element is found, the whole column and row
+    will be set to 0.
+    Runtime: O(M * N)
+    Space Complexity: O(min(M, N))
+    :param matrix: an MxN matrix.  M is the number of rows, N is the number of columns
+    :return: a zero-modified matrix
+    """
+    M = len(matrix)
+    N = len(matrix[0])
+
+    zeroed_rows = set()
+    zeroed_cols = set()
+
+    for row in range(M):
```

Comment:
you can also write this as

```suggestion
    for i, row in enumerate(M):
        for j, num in enumerate(row):
```

which is more internally consistent.